### ISSUE TITLE
About other languages

### ISSUE THREAD

**Issue by piano777wzy** [CG8516/PvZA11y#16]
Hello, I have an idea about other languages.
I don't know if we can add the function of allowing users to translate the language themselves. For example, if I use Chinese now, if I use this auxiliary mod, I have to use a translation tool for translation, so that if one day my network is gone, or the translation is broken, it will be more troublesome. I believe there are many people from other countries who also need to read the text aloud in their own language. Such as Japan, Brazil, Portugal, China and so on.
You said earlier that it will take a lot of time to support more versions, and there are so many other language versions of this game in the world that there are more than 3 Chinese versions of our country's annual version.
So there is no way to provide a language template, it can be a file .txt, of course, it can also be in other formats.
I don't know if you have ever used a software called reaper. Although it is not a game, it provides a language template that all users in the world can translate and choose the language in the option.
Well, let me say something else, and there is also a game called Crazy Party, which also provides a language interpreter.
Of course, I'm not a programmer, and I haven't learned programming, so I don't know if it's difficult to implement.
Such a function would be great, so I can translate it once and share it with others. After the update, you just need to add the updated language entry to the template and let the users translate it.

**Comment by guilevi** [comment]
As far as I'm aware, localization for PVZ is kind of messy—I was only able to find a few replacements for the main.pak file which supposedly do it. They do indeed translate the on-screen text and the mod does read it, but a lot of text comes from the mod itself and that would have to be localized separately. For languages with accent marks (and likely other unicode characters) they also don't currently read properly.

**Comment by CG8516** [comment]
It's definitely something I want to support, but yes, it's messy.  

I'll need to rework a few things, and move all the mod text to external files.  
I was thinking of having some sort of dynamic language system, where you could copy a new language to the mod's folder, and it will automatically detect it, and add it to a list of lanaguages you can choose from.
That way, anybody could make new translations without needing to touch the code.  

I won't make any promises, but I'm hoping to get it done before the end of the year.

**Comment by CG8516** [comment]
Added this in beta 1.15! Depending on your timezone, I got it done before the end of the year, haha.  
To create a language, make a copy of the 'English' folder, name it your language, and replace the text within the .yaml files with whatever language text you want.  

Yaml files are basically just .txt files, and can be edited with notepad.  
They're pretty friendly to edit, just make sure you only edit the contents after the colon ':'  
Also make sure you keep all blank lines, and special characters, like '- >-' and '- '
